### PR TITLE
Make failed exception handling more verbose

### PR DESCRIPTION
**Pull request by prp**
This PR generates a descriptive warning when an exception cannot be handled inside of the enclave.

This should help debug some of the exception failures when running LTP tests.

### CODE REVIEW

File: tests/ltp/ltp-batch2/ltp_disabled_tests.txt [hunk 1]
```diff
@@ -800,9 +800,9 @@
 #/ltp/testcases/kernel/syscalls/sendfile/sendfile08
 #/ltp/testcases/kernel/syscalls/sendfile/sendfile09
 #/ltp/testcases/kernel/syscalls/sendmmsg/sendmmsg01
-#/ltp/testcases/kernel/syscalls/sendmsg/sendmsg01
+/ltp/testcases/kernel/syscalls/sendmsg/sendmsg01
```
Comment (hukoyu) [inline tests/ltp/ltp-batch2/ltp_disabled_tests.txt]: Why are you disabling these test cases? This time these tests failed randomly, next time different tests can fail. I think we shouldn't disable them.

File: tests/ltp/ltp-batch1/ltp_disabled_tests.txt [hunk 2]
```diff
@@ -261,10 +261,10 @@
 #/ltp/testcases/kernel/syscalls/fstatat/fstatat01
 #/ltp/testcases/kernel/syscalls/fstatfs/fstatfs01
 #/ltp/testcases/kernel/syscalls/fstatfs/fstatfs02
-#/ltp/testcases/kernel/syscalls/fsync/fsync01
-#/ltp/testcases/kernel/syscalls/fsync/fsync02
-#/ltp/testcases/kernel/syscalls/fsync/fsync03
-#/ltp/testcases/kernel/syscalls/fsync/fsync04
+/ltp/testcases/kernel/syscalls/fsync/fsync01
```
Comment (hukoyu) [inline tests/ltp/ltp-batch1/ltp_disabled_tests.txt]: Why are we disabling fsync tests? We are loosing total syscall test coverage here. As we discussed we cannot use loop device file system and we use root file system to enable these tests. What solution do you recommend to run these  tests?

**Review by hukoyu — CHANGES_REQUESTED** [review]
I don't think we should disable these tests. We are loosing test coverae here. Also due to some issues tests fail randomly. This time it was sendmsg and sendto but it doesn't mean they will fail next time. Shall we disable every randomly failed test due to issues we haven't figued out root cause? Also if you disable these tests, how shall we see the additional info you ar eadding in this PR?
cc @paulcallen @Pengpeng-Microsoft @mikbras

**Comment by prp** [comment]
@hukoyu As I explained, these tests are fundamentally broken:
- `fsync` relies on block devices being created inside the enclave, which is not possible currently.
- `sendto`/`sendmsg` want to create a client/server setup inside the enclave. This requires multiple processes, which is also not possible.

These tests therefore do not test anything meaningful, and they should have never been merged in the first place.

**Comment by paulcallen** [comment]
We need to stop disabling tests that blow holes in our testing. If they are flakey then we need to fix the tests. Disabling at this point is not acceptable unless we get consensus across the team that we have a contingency plan to re-enable it.

**Review by paulcallen — CHANGES_REQUESTED** [review]
We need to stop disabling tests that blow holes in our testing. If they are flakey then we need to fix the tests. Disabling at this point is not acceptable unless we get consensus across the team that we have a contingency plan to re-enable it.

**Comment by prp** [comment]
@paulcallen the LTP tests that I disabled were not flakey but incorrect. We need better QA and not add incorrectly patched LTP tests to our CI pipeline.

**Review by davidchisnall — APPROVED** [review]
Looks good to me, but it might be better to pull the test suite changes into a different PR. I agree that we shouldn’t be trying to run LTP tests that test features that will not work with SGX. I am a bit surprised that here we’re passing previously, that implies a need for better QA before enabling an LTP test.

**Comment by prp** [comment]
> Looks good to me, but it might be better to pull the test suite changes into a different PR. I agree that we shouldn’t be trying to run LTP tests that test features that will not work with SGX. I am a bit surprised that here we’re passing previously, that implies a need for better QA before enabling an LTP test.

@davidchisnall I've moved the disabling of incorrect LTP tests to a separate PR https://github.com/lsds/sgx-lkl/pull/549.

File: tests/ltp/ltp-batch2/ltp_disabled_tests.txt (hunk 1 above)
Comment (prp) [reply tests/ltp/ltp-batch2/ltp_disabled_tests.txt]: I've provided an explanation, and moved the changes to the CI pipeline to a separate PR.

File: tests/ltp/ltp-batch1/ltp_disabled_tests.txt (hunk 2 above)
Comment (prp) [reply tests/ltp/ltp-batch1/ltp_disabled_tests.txt]: I've provided an explanation, and moved the changes to the CI pipeline to a separate PR.

**Review by hukoyu — APPROVED** [review]
DIsabling test moved to another PR. This PR looks good.

**Review by paulcallen — APPROVED** [review]

**Pull request merged**